decision_rules/core/ruleset.py

```diff
@@ -52,22 +40,26 @@ def __init__(
         self._stored_default_conclusion: AbstractConclusion = None
         self._prediction_strategy: Optional[PredictionStrategy] = None
         self.decision_attribute: Optional[str] = None
-        self._voting_weights_calculated: bool = False
 
     @abstractmethod
     def get_metrics_object_instance(self) -> AbstractRulesMetrics:
         """Returns metrics object instance."""
         return None
 
-    def _validate__object_state_before_prediction(self):
-        if not self._voting_weights_calculated:
+    def _validate_object_state_before_prediction(self):
+        voting_weights_calculated: bool = all(
+            [rule.coverage is not None for rule in self.rules]
+        )
+        if not voting_weights_calculated:
             raise InvalidStateError(
-                'Rules coverages must have been calculated before prediction.' +
-                'Did you forget to call update(...) method?'
+                "Rules coverages must have been calculated before prediction."
```

Comment:
To w zasadzie językowa uwaga. "Must have been" sugeruje inne znaczenie, niż chcemy. Zmieniłbym to na:
"Rule coverages have to be calculated before performing prediction."

Reply:
[fix: Code review requested changes](https://github.com/ruleminer/decision-rules/pull/15/commits/78eb3b39e6c5a3e80c96b8e8dab81874bd5bc49a)